### PR TITLE
PWA-3154:GraphQL cacheable requests always have Authorization Bearer …

### PR DESCRIPTION
**Pull request by glo11372**
## Description

GraphQL cacheable requests always have "Authorization Bearer" header
Steps to test the behaviour :

1- sign up.in Venia.magento.com or any PWA store
2- login
3- check network tab on any page and search for GraphQL requests which should be cached such as CMSPages
4- check the request header you will not find Authorization Bearer header exist
5- Varnish will cache that request's response.

## Related Issue

Closes # https://jira.corp.adobe.com/browse/PWA-3154

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/Apollo/links/__tests__/index.spec.js [hunk 1]
```diff
@@ -7,10 +7,10 @@ jest.mock('@apollo/client', () => ({
     __esModule: true,
     createHttpLink: jest.fn(() => 'http')
 }));
-jest.mock('@magento/peregrine/lib/Apollo/links/authLink', () => ({
-    __esModule: true,
-    default: jest.fn(() => 'auth')
-}));
+// jest.mock('@magento/peregrine/lib/Apollo/links/authLink', () => ({
```
Comment (glo82145) [inline packages/peregrine/lib/Apollo/links/__tests__/index.spec.js:10]: Please donot comment line like these .. remove it from code

File: packages/peregrine/lib/Apollo/links/index.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { createHttpLink } from '@apollo/client';
 
-import createAuthLink from '@magento/peregrine/lib/Apollo/links/authLink';
+// import createAuthLink from '@magento/peregrine/lib/Apollo/links/authLink';
```
Comment (glo82145) [inline packages/peregrine/lib/Apollo/links/index.js:3]: remove code

Comment (glo11372) [reply packages/peregrine/lib/Apollo/links/index.js]: Done

**Review by glo82145 — APPROVED** [review]

**Comment by fooman** [comment]
How will signed in customer functionality work with the authLink removed?
https://github.com/magento/pwa-studio/blob/d12b8c40678a43d3ba173f7eab2509f7687caa65/packages/peregrine/lib/Apollo/links/authLink.js#L15

**Comment by glo85315** [comment]
> How will signed in customer functionality work with the authLink removed?



> How will signed in customer functionality work with the authLink removed?
> 
> https://github.com/magento/pwa-studio/blob/d12b8c40678a43d3ba173f7eab2509f7687caa65/packages/peregrine/lib/Apollo/links/authLink.js#L15

Screenshot 2024-03-27 at 6 27 201PM
 Hi this will work after removing  **authLink** .because when user logined server will create cookies and send back to client side. in client side cookies will store in user browser and this cookies will take in to next subsequent API Header request as cookies. then server validate cookie and if its valid it will allow the authetication.


* Authentication: When a user logs in to your application, your server typically creates a session for that user and issues a session identifier in the form of a cookie. This cookie is then stored in the user's browser.
* 		Subsequent Requests: When the user performs actions or accesses protected resources that require authorization, your React application sends requests to the server. The browser automatically includes the cookies associated with the domain in the request headers.
* 		Server-side Authentication: On the server side, your application checks for the presence of the session cookie in the request headers. If the session cookie is present and valid, the server identifies the user associated with the session and authorizes the request accordingly.

**Comment by glo82145** [comment]
deploy pr-test

**Comment by glo82145** [comment]
deploy pwa-pr-test

**Review by glo82145 — APPROVED** [review]

**Pull request merged**